src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/BufferBlockCopyLengthAnalyzer.cs

```diff
@@ -0,0 +1,179 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.BufferBlockCopyDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Reliability,
+                                                                                      RuleLevel.IdeSuggestion,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemBuffer, out INamedTypeSymbol? bufferType))
+                {
+                    return;
+                }
+
+                IMethodSymbol blockCopyMethod = bufferType
+                   .GetMembers("BlockCopy")
+                   .OfType<IMethodSymbol>()
+                   .FirstOrDefault();
+
+                if (blockCopyMethod is null)
+                {
+                    return;
+                }
+
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemArray, out INamedTypeSymbol? arrayType))
+                {
+                    return;
+                }
+
+                IPropertySymbol arrayLengthProperty = arrayType
+                    .GetMembers("Length")
+                    .OfType<IPropertySymbol>()
+                    .FirstOrDefault();
+
+                if (arrayLengthProperty is null)
+                {
+                    return;
+                }
+
+                INamedTypeSymbol byteType = context.Compilation.GetSpecialType(SpecialType.System_Byte);
+                INamedTypeSymbol sByteType = context.Compilation.GetSpecialType(SpecialType.System_SByte);
+
+                if (byteType is null || sByteType is null)
+                {
+                    return;
+                }
```

Comment:
This is definitely redundant, we don't need to load these symbols, for type check you can just use the SpeicalType enums https://sourceroslyn.io/#Microsoft.CodeAnalysis/SpecialType.cs,78